packages/amplify-provider-awscloudformation/lib/update-idp-roles-cfn.json

```diff
@@ -0,0 +1,173 @@
+{
+
+    "UpdateRolesWithIDPFunction": {
+        "DependsOn": [
+            "AuthRole",
+            "UnauthRole"
+        ],
+        "Type": "AWS::Lambda::Function",
+        "Properties": {
+            "Code": {
+                "ZipFile": {
+                    "Fn::Join": [
+                        "\n",
+                        [
+                            "const response = require('cfn-response');",
+                            "const aws = require('aws-sdk');",
+                            "let responseData = {};",
+                            "exports.handler = function(event, context) {",
+                            "  try {",
+                            "    let authRoleName = event.ResourceProperties.authRoleName;",
+                            "    let unauthRoleName = event.ResourceProperties.unauthRoleName;",
+                            "    let idpId = event.ResourceProperties.idpId;",
+                            "    let promises = [];",
+                            "    let authParamsJson = { 'Version': '2012-10-17','Statement': [{'Effect': 'Allow','Principal': {'Federated': 'cognito-identity.amazonaws.com'},'Action': 'sts:AssumeRoleWithWebIdentity','Condition': {'StringEquals': {'cognito-identity.amazonaws.com:aud': idpId},'ForAnyValue:StringLike': {'cognito-identity.amazonaws.com:amr': 'authenticated'}}}]};",
+                            "    let unauthParamsJson = { 'Version': '2012-10-17','Statement': [{'Effect': 'Allow','Principal': {'Federated': 'cognito-identity.amazonaws.com'},'Action': 'sts:AssumeRoleWithWebIdentity','Condition': {'StringEquals': {'cognito-identity.amazonaws.com:aud': idpId},'ForAnyValue:StringLike': {'cognito-identity.amazonaws.com:amr': 'unauthenticated'}}}]};",
```

Comment:
Can we move these to separate lines for readability? Alternatively perhaps you could build this up so that it's easier to add more conditions in the future. Something like this:

```
const authParamsJson = ['condition1',
  'condition2']
  .join('');
```

Of course you'll need to substitute in the curly braces but it might be easier to have this as a builder for the future.